app/cdap/components/PipelineHistory/index.tsx

```diff
@@ -0,0 +1,171 @@
+import styled from 'styled-components';
+import T from 'i18n-react';
+import { PipelineHistoryTable } from './PipelineHistoryTable';
+import { useQuery } from '@apollo/react-hooks';
+import PaginationStepper from 'components/shared/PaginationStepper';
+import { Provider, useSelector } from 'react-redux';
+import Store, {
+  nextPage,
+  prevPage,
+  QUERY,
+  setPageLimit,
+  setVersions,
+} from './PipelineHistoryStore';
+import SelectWithOptions from 'components/shared/SelectWithOptions';
+import { LoadingAppLevel } from 'components/shared/LoadingAppLevel';
+import { PipelineHistoryTableDiv } from './styles';
+import { MyPipelineApi } from 'api/pipeline';
+
+const PREFIX = 'features.PipelineHistory';
+
+interface IPipelineHistoryProps {
+  isOpen: boolean;
+  toggle: () => void;
+  anchorEl: any;
+  pipelineName: string;
+}
+
+const PaginationContainer = styled.div`
+  margin-right: 20px;
+  display: block;
+  div {
+    display: inline-block;
+    margin-right: 10px;
+  }
+`;
+
+const PipelineHistory = ({ isOpen, toggle, anchorEl, pipelineName }: IPipelineHistoryProps) => {
+  const { ready, pageToken, pageLimit, pipelineVersions } = useSelector(({ versions }) => versions);
+  const [isRestoreLoading, setIsRestoreLoading] = useState(false);
+  const [latestVersion, setLatestVersion] = useState(null);
+
+  const { loading, error, data, refetch, networkStatus } = useQuery(QUERY, {
+    errorPolicy: 'all',
+    fetchPolicy: 'no-cache',
+    notifyOnNetworkStatusChange: true,
+    variables: {
+      nameFilter: pipelineName || undefined,
+      token: pageToken || undefined,
+      pageSize: pageLimit,
+      namespace: getCurrentNamespace(),
+      orderBy: 'DESC',
+      nameFilterType: 'EQUALS',
+    },
+  });
+
+  const Pagination = ({}) => {
+    const { prevDisabled, nextDisabled, pageCount, pageIndex, pageLimitOptions } = useSelector(
+      ({ versions }) => ({
+        prevDisabled: !versions.previousTokens.length,
+        nextDisabled: !versions.nextPageToken,
+        pageCount: versions.pipelineVersions.length,
+        pageIndex: versions.previousTokens.length,
+        pageLimitOptions: versions.pageLimitOptions,
+      })
+    );
+
+    const onChange = (e) => {
+      setPageLimit(e.target.value);
+    };
+
+    return (
+      <PaginationContainer className="float-right">
+        <div>{T.translate(`${PREFIX}.table.rowsPerPage`)}</div>
+        <div>
+          <SelectWithOptions value={pageLimit} onChange={onChange} options={pageLimitOptions} />
+        </div>
+        <div>{`${pageIndex * pageLimit + 1} - ${pageIndex * pageLimit + pageCount}`}</div>
+        <div>
+          <PaginationStepper
+            onNext={nextPage}
+            onPrev={prevPage}
+            nextDisabled={nextDisabled}
+            prevDisabled={prevDisabled}
+          />
+        </div>
+      </PaginationContainer>
+    );
+  };
+
+  useEffect(() => {
+    if (loading || networkStatus === 4) {
```

Comment:
Can you import NetworkStatus and do `networkStatus === NetworkStatus['whatever4is']`